Refactor for PSR-4

XedinUnknown opened the pull request:
⚠️ Requires #233.

This extracts the changes related to PSR-4 refactoring from #120.

- Add package configuration to every module, and includes them all in main project - a more distributed modular system.
- Fix some class names.
- Project now fully complies with PSR-4.
- Update some constraints, and upgrade dependencies.
- Platform requirements now defined - requires PHP 7.1.

XedinUnknown committed:
Now depending on newer module standard

Refactoring of modules themselves was not necessary,
because they were already following the new standard.

XedinUnknown committed:
Use Composer modules and convert modules to PSR-4

PSR-4 is much more robust and predictable. But to do this,
a source root dir must be specified for every module.
This could be done in the root file, but this is not very modular.
Instead, now every module declares its own source root by using
the amazing Composer Merge Plugin. This approach allows each module
to also declare its own dependencies. Together, these changes allow
modules to be easily extractable to separate pacakges when the need
arises, and in general improves modularity significantly.

XedinUnknown committed:
Remove unused dependency

The WP containers don't seem to be used anywhere. There were
imports, but they are not actually used it seems.

XedinUnknown committed:
No longer requiring WPCS for file names

XedinUnknown committed:
Configure PHPCS

Now will use config instead of CLI values for most things.
Will also show sniff codes.

XedinUnknown committed:
Add Container typehint to all service definitions

XedinUnknown committed:
Now using modularity standard v0.3

This is the newest release, and is the first one that is compatible
with PHP 8. Other changes are BC-breaking, but very very minor.

XedinUnknown committed:
Merge pull request #2 from strangerkir/task/refactoring

XedinUnknown committed:
Revert "Switch GPG keyserver"

This reverts commit 25e419ab

XedinUnknown committed:
Switch GPG keyserver

https://stackoverflow.com/a/68132500/565229

XedinUnknown committed:
Upgrade Composer

XedinUnknown committed:
Upgrade Node

XedinUnknown committed:
Use Composer v1

XedinUnknown committed:
Update deps

XedinUnknown committed:
Bring remaining classes to comply with PSR-4

XedinUnknown committed:
Add package definition to module

AlexP11223 commented on .github/workflows/php.yml:
```diff
@@ -1,27 +1,25 @@
+          php-version: ${{ matrix.php-versions }}
+
+    - name: Packagist.com Auth
+      run: composer config --global --auth http-basic.repo.packagist.com token ${{ secrets.PACKAGIST_TOKEN }}
```
Do we need this here? I think all packages are public.

XedinUnknown replied:
No problem to remove, if extra. It was just a useful thing from the source project.

AlexP11223 commented on .github/workflows/php.yml:
```diff
@@ -31,7 +29,8 @@ jobs:
+      run: vendor/bin/phpunit
+
+    - name: Run PHPCS
+      run: ./vendor/bin/phpcs--runtime-set ignore_warnings_on_exit 1 src modules woocommerce-paypal-payments.php --extensions=php
```
Why ignore warnings? Currently there are no warning. :)

XedinUnknown replied:
Because whether they are or aren't there, they should not cause the task to fail.

AlexP11223 replied:
Why? It makes sense to ignore warnings if phpcs is added to an old project with lots of warnings, but here we have no warnings right now, so why not keep it this way? 

If there are bad warning rules, then these rules should be disabled, not ignored, because this will result in polluted list of warnings and it will be difficult to see new warnings.

Also because of the weird WP indent rules we often use `phpcbf` here, so if we ignore warnings it may result in lots of changes of unrelated files, making PRs difficult to read.

XedinUnknown replied:
Because a Warning severity does not indicate failure. It indicates something suspicious that should be brought to our attention, but not something that is outright wrong. In other words, our PHPCS standards are a specification, and if there are only warnings, it means that we still comply with it. But we should still see warnings, as they indicate that something _may_ be wrong.

`phpcbf` and other auto-fixing stuff is just there to help. I always try to commit those changes separately. If you are talking about indenting rules, then I guess you are talking about whitespace. Github PRs have the [option to hide whitespace changes][1], so this is not reallhy a problem.

[1]: https://github.blog/2018-05-01-ignore-white-space-in-code-review/

AlexP11223 commented on docker/Dockerfile_wp:
```diff
@@ -74,9 +74,18 @@ RUN docker-php-ext-install intl json && \
+RUN curl -sL https://deb.nodesource.com/setup_16.x | bash -
 RUN apt-get install -y nodejs
 
+# Install Phive
```
Why install phive if not used?

XedinUnknown replied:
It's another thing from the source project that I thought would be useful. No problem to remove, if not needed.

XedinUnknown replied:
Removed in 1866efc.

AlexP11223 commented on docker/Dockerfile_wp:
```diff
@@ -64,7 +64,7 @@ RUN apt-get install -y \
 
 RUN php -r "copy('https://getcomposer.org/installer', 'composer-setup.php');" && \
-    php composer-setup.php --install-dir=/usr/bin --filename=composer --version=1.10.19 && \
+    php composer-setup.php --install-dir=/usr/bin --filename=composer --version=1.10.22 && \
```
I think it can be just 2 (default).

XedinUnknown replied:
It cannot, unfortunately, due to Composer plugin dependency constraints.

`wikimedia/composer-merge-plugin:1.4` requires `composer-plugin-api:^1.0`. `wikimedia/composer-merge-plugin:^2.0.1` supports `composer-plugin-api:^2.0`, but requires PHP 7.2, which is higher than this project's minimal requirement.

XedinUnknown committed:
Add missing module Composer config

XedinUnknown committed:
Use Composer v1 on CI

XedinUnknown committed:
Fix typo in command

XedinUnknown committed:
Auto-fix PHPCS

XedinUnknown committed:
Update parameter name

Fixes PHPCS

XedinUnknown committed:
Update PHPStorm configs

XedinUnknown committed:
PHPCS now allows omitting inherited param doc

Dinamiko commented:
Hi @jorgeatorres before merging this PR I need your feedback on renaming files to follow PSR-4 convention, we firmly believe this is a good move but we´ll need your OK before merging it, when you have time, please let us know which are your thought or if you see any potential problem in doing so, thanks.

AlexP11223 committed:
Fix docker phpcs command

do not hide warnings

AlexP11223 committed:
Make command style in readme more consistent

AlexP11223 committed:
Use yarn for building package, and improve readme

to not require npm too

XedinUnknown committed:
Merge branch 'bug/fix-project-configuration' into task/refactor-for-psr-4

# Conflicts:
#	.github/workflows/php.yml
#	.idea/php.xml
#	.idea/woocommerce-paypal-payments.iml

jorgeatorres commented on modules/ppcp-compat/src/PPEC/PpecHelper.php:
```diff
@@ -11,7 +11,7 @@
  * out the status of the gateway.
  */
-class PPECHelper {
+class PpecHelper {
```
Would something break if we keep this class as `PPECHelper`. Keep in mind that `PPEC` is an acronym (PayPal Express Checkout) and it looks a bit weird capitalized that way.
Alternatively, we could just rename it to `PayPalCheckoutHelper` or something like that.

XedinUnknown replied:
✔️ 40b5783

jorgeatorres commented on modules/ppcp-onboarding/src/OnboardingRestController.php:
```diff
@@ -16,7 +16,7 @@
  * Exposes and handles REST routes related to onboarding.
  */
-class Onboarding_REST_Controller {
+class OnboardingRestController {
```
Similar to the above, REST is an acronym.

XedinUnknown replied:
✔️ 9f2496a

jorgeatorres commented:
>  I need your feedback on renaming files to follow PSR-4 convention, we firmly believe this is a good move but we´ll need your OK before merging it

@Dinamiko: I believe we'll have to ignore some naming rules in the PHPCS config file, but I'm ok with this change. This convention is already being used in other WooCommerce packages so it wouldn't be the first time we deviate from the naming convention in the WooCommerce standards.

That said, and given this PR contains a ~180 files changeset, is there anything else I should be taking a closer look at?
I added a couple of comments about instances of certain acronyms (such as REST and PPEC) being capitalized differently, which I believe is odd, but not sure if it's needed for the autoloader to work or something. Please let me know 😃.

Dinamiko commented:
@jorgeatorres thanks for the review,

My idea was not to assing the PR to you because it contains a lot of files :) it was just to know your thoughs on PSR-4 renaming.

Regarding acronyms I´ve just discused with the team and there is nothing wrong on using them, nothing will be broken, it was just done this way to follow the standard strictly, using capitalization for word separation.

So let´s use acronyms for these two (REST and PPEC) and I´ll check if there are more there.

Dinamiko commented:
@XedinUnknown PR looks good, just a little adjustment about acronyms on class and file names and we are good to merge:
When you have time, can you please rename these:
-  `PpecHelper` to `PPECHelper` 
- `OnboardingRestController` to `OnboardingRESTController`
- `DccProductStatus` to `DCCProductStatus`
- `WcGatewayModule` to `WCGatewayModule` or `WooCommerceGatewayModule`

Dinamiko committed:
Do not throw error if customer does not have payment tokens

Dinamiko committed:
Disable credit card fields when selecting saved payment

Dinamiko committed:
Refactoring

AlexP11223 committed:
Refactor webhook registration errors logging

Logging was useless because it was just repeating the exception message. Context args were not used.

Dinamiko committed:
Log request and response information

Dinamiko committed:
Log request body and wp error

XedinUnknown committed:
Merge branch 'bug/fix-project-configuration' into task/refactor-for-psr-4

Dinamiko committed:
Refactoring

XedinUnknown committed:
Change case of `PPECHelper` class name

XedinUnknown committed:
Change case of `OnboardingRestController` class name

XedinUnknown committed:
Change case of `DCCProductStatus` class name

XedinUnknown committed:
Change case of `WCGatewayModule` class name

See https://github.com/woocommerce/woocommerce-paypal-payments/pull/235#issuecomment-910082924.

XedinUnknown commented:
I confirm that this kind of class names are explicitly allowed by point 2.2.5 of [PSR-4][]. The requested classes have been renamed as requested, in 40b5783, 9f2496a, 6078de2, and 72216ba.

[PSR-4]: https://www.php-fig.org/psr/psr-4/

Dinamiko committed:
Log into single entry

AlexP11223 committed:
Fix and refactor credential change handling

AlexP11223 committed:
Add logging

AlexP11223 committed:
Register webhooks a bit later

Otherwise it uses old state (and cannot choose the correct host, etc.), at least with ALTERNATE_WP_CRON

Dinamiko committed:
Add and remove class for css opacity

AlexP11223 committed:
Do not detect credentials change on DCC settings page

Dinamiko committed:
Dispatch hosted fields loaded event

Dinamiko committed:
Add `-disabled` to css class name

Dinamiko committed:
Do not log sensible body data

Dinamiko committed:
Move request response information from logger to request trait and return it as string

Dinamiko committed:
Fix unit tests (WIP)

jorgeatorres committed:
Update wp.org assets

jorgeatorres committed:
Merge pull request #258 from woocommerce/update-assets

Update wp.org assets

Dinamiko committed:
Fix unit tests

Dinamiko committed:
Remove unused class

Dinamiko committed:
Remove unused class

Dinamiko committed:
Merge pull request #253 from woocommerce/PCP-287-improve-logging

Improve logging

Dinamiko committed:
Add vaulting module

AlexP11223 committed:
Merge branch 'trunk' into fix-webhooks-auth

Dinamiko committed:
Merge pull request #255 from woocommerce/fix-webhooks-auth

Fix webhooks auth

AlexP11223 committed:
Implement webhook list retrieval

AlexP11223 committed:
Refactor field page checking

Reduce code duplication and support more than 2 pages.
Also remove 'all'

AlexP11223 committed:
Add table field renderer

AlexP11223 committed:
Support passing field value without settings

Pass as function to not perform e.g. HTTP requests if the field is not needed.

AlexP11223 committed:
Add webhooks status page

Dinamiko committed:
Disable hosted fields when selecting saved credit card

Dinamiko committed:
Fix merge conflict

AlexP11223 committed:
Show webhooks

AlexP11223 committed:
Improve table style

AlexP11223 committed:
Change "register" to "subscribe" for consistency

AlexP11223 committed:
Try to subscribe if no webhooks when visiting webhooks status page

Dinamiko committed:
Merge pull request #251 from woocommerce/PCP-244-saved-credit-card-does-not-auto-

Improve UX for stored credit card payments

AlexP11223 committed:
Add resubscribe webhooks button

AlexP11223 committed:
Fix name in package.json

Dinamiko committed:
Merge pull request #259 from woocommerce/webhooks-status-page

Webhooks status page

Dinamiko committed:
Bump 1.6.0 version

Dinamiko committed:
Update PayPal JSSDK integration date

Dinamiko committed:
Merge branch 'trunk' into PCP-288-vaulting-token-creation-fails

Dinamiko committed:
Display saved payment tokens

Dinamiko committed:
Move payment tokens repository to vaulting package

Dinamiko committed:
Check if hosted fields instance exist before setting attribute

Dinamiko committed:
Do not display generic error on hosted fields submit. Fixes #263.

Dinamiko committed:
Revert `e7a4dbd5061ccc773293077f9ed5afefbbcb958e`

Dinamiko committed:
Do not set shipping if country does not have postal code

Dinamiko committed:
Log 3DS authentication result

Dinamiko committed:
Fix phpunit test

AlexP11223 committed:
Add service for current webhook

AlexP11223 committed:
Fix log message

Dinamiko committed:
Work in progress

Dinamiko committed:
Add js script for my account payments screen

AlexP11223 committed:
Add webhook simulation

AlexP11223 committed:
Make event type param

AlexP11223 committed:
Add test

Dinamiko committed:
Load my account payments script only in its own page

Dinamiko committed:
Add delete payment functionality

AlexP11223 committed:
Log endpoint errors

Dinamiko committed:
Merge pull request #275 from woocommerce/log-errors

Log endpoint errors

Dinamiko committed:
Merge branch 'trunk' into release/1.6.0

Dinamiko committed:
Log delete payment failure and parse url when checking payments page

Dinamiko committed:
Flush rewrite rules on plugin activation to make my account payments screen available

Dinamiko committed:
Fix phpcs errors

Dinamiko committed:
Merge pull request #273 from woocommerce/webhook-simulation

Webhook simulation

Dinamiko committed:
Merge pull request #274 from woocommerce/PCP-288-vaulting-token-creation-fails

Allow customers to see and delete their saved payments in My Account.

Dinamiko committed:
Merge branch 'trunk' into release/1.6.0

Dinamiko committed:
Update changelog

Dinamiko committed:
Update changelog

Dinamiko committed:
Add logger to data client id endpoint service

Dinamiko committed:
Add logger to pending services

Dinamiko committed:
Make webhook nullable to prevent fatal error

Dinamiko committed:
Fix typo in class name

jorgeatorres committed:
Merge pull request #261 from woocommerce/release/1.6.0

Sync `release/1.6.0` fixes into `trunk`.

AlexP11223 committed:
Use CHECKOUT.ORDER.APPROVED for simulation

Not sure if it matters which event we use, but at least we are subscribed to this one

AlexP11223 committed:
Add test

Dinamiko committed:
Add authorization flow for saved credit cards

Dinamiko committed:
Merge pull request #277 from woocommerce/change-webhook-simulation-event

Use CHECKOUT.ORDER.APPROVED for simulation

Dinamiko committed:
Fix phpcs errors

Dinamiko committed:
Merge pull request #287 from woocommerce/PCP-302-using-a-saved-card-it-showed-an-

Add authorization flow for saved credit cards

Dinamiko committed:
Regenerate `yarn.lock` files

Dinamiko committed:
Update all packages to latest versions

Dinamiko committed:
Use `sass` instead of `node-sass`

Dinamiko committed:
Merge pull request #291 from woocommerce/PCP-312-dependabots-issues

Update JS packages

Dinamiko committed:
Update packages on vaulting module

Dinamiko committed:
Merge pull request #292 from woocommerce/update-webpack-on-vaulting-module

Update packages on vaulting module

jorgeatorres committed:
Remove node_modules/ directories before building ZIP

jorgeatorres committed:
Update release date in changelog

AlexP11223 committed:
Merge branch 'trunk' into fix-cmd

Dinamiko committed:
Merge pull request #233 from woocommerce/bug/fix-project-configuration

Fix Project Configuration

Dinamiko commented:
Reopening PR

Dinamiko committed:
Merge pull request #237 from woocommerce/fix-cmd

Fix docker phpcs command and improve readme

Dinamiko committed:
Fix merge conflicts

Dinamiko committed:
add local changes

Dinamiko committed:
Merge conflicts
